crates/apub/src/api/resolve_object.rs

```diff
@@ -67,14 +71,154 @@ async fn convert_response(
+  use lemmy_utils::{error::LemmyResult, LemmyErrorType};
+  use serial_test::serial;
+
+  async fn create_user(
+    instance_id: InstanceId,
+    name: String,
+    admin: bool,
+    context: &Data<LemmyContext>,
+  ) -> LemmyResult<LocalUserView> {
+    let person_form = PersonInsertForm::test_form(instance_id, &name);
+    let person = Person::create(&mut context.pool(), &person_form).await?;
+
+    let user_form = match admin {
+      true => LocalUserInsertForm::test_form_admin(person.id),
+      false => LocalUserInsertForm::test_form(person.id),
+    };
+    let local_user = LocalUser::create(&mut context.pool(), &user_form, vec![]).await?;
+
+    Ok(LocalUserView::read(&mut context.pool(), local_user.id).await?)
+  }
+
+  #[tokio::test]
+  #[serial]
+  #[expect(clippy::unwrap_used)]
+  async fn test_object_visibility() -> LemmyResult<()> {
+    let context = LemmyContext::init_test_context().await;
+
+    let instance = Instance::read_or_create(&mut context.pool(), "example.com".to_string()).await?;
+
+    let site_form = SiteInsertForm::new("test site".to_string(), instance.id);
+    let site = Site::create(&mut context.pool(), &site_form).await?;
+
+    let local_site_form = LocalSiteInsertForm {
+      site_setup: Some(true),
+      private_instance: Some(false),
+      ..LocalSiteInsertForm::new(site.id)
+    };
+    LocalSite::create(&mut context.pool(), &local_site_form).await?;
+
+    let creator = create_user(instance.id, "creator".to_string(), false, &context).await?;
+    let regular_user = create_user(instance.id, "user".to_string(), false, &context).await?;
+    let admin_user = create_user(instance.id, "admin".to_string(), true, &context).await?;
+
+    //let community_insert_form = ;
+    let community = Community::create(
+      &mut context.pool(),
+      &CommunityInsertForm::new(
+        instance.id,
+        "test".to_string(),
+        "test".to_string(),
+        "pubkey".to_string(),
+      ),
+    )
+    .await?;
+
+    let post_insert_form = PostInsertForm::new("Test".to_string(), creator.person.id, community.id);
+    let post = Post::create(&mut context.pool(), &post_insert_form).await?;
+
+    let query = format!("q={}", post.ap_id).to_string();
+    let query: Query<ResolveObject> = Query::from_query(&query)?;
+
+    // Objects should be resolvable without authentication
+    let res = resolve_object(query.clone(), context.reset_request_count(), None).await?;
+    assert_eq!(res.post.as_ref().unwrap().post.ap_id, post.ap_id);
+    // Objects should be resolvable by regular users
+    let res = resolve_object(
+      query.clone(),
+      context.reset_request_count(),
+      Some(regular_user.clone()),
+    )
+    .await?;
+    assert_eq!(res.post.as_ref().unwrap().post.ap_id, post.ap_id);
+    // Objects should be resolvable by admins
+    let res = resolve_object(
+      query.clone(),
+      context.reset_request_count(),
+      Some(admin_user.clone()),
+    )
+    .await?;
```

Comment:
No need for this check.

Reply:
as this is in rust tests it should take a fraction of a second, i put it in there mostly for having the tests be the same 3 set of checks as for the deleted item. if you want it removed i can remove it.

Reply:
Thats true but imo it makes the test code unnecessarily long. No real harm in keeping it though.